src/compiler/checker.ts

```diff
@@ -27440,104 +27787,257 @@ namespace ts {
         }
 
         /**
-         * This function has very similar logic as getIteratedTypeOfIterable, except that it operates on
-         * Iterators instead of Iterables. Here is the structure:
+         * Gets the *yield*, *return*, and *next* types from an `Iterator`-like or `AsyncIterator`-like type.
          *
-         *  { // iterator
-         *      next: { // nextMethod
-         *          (): { // nextResult
-         *              value: T // nextValue
-         *          }
-         *      }
-         *  }
+         * If we successfully found the *yield*, *return*, and *next* types, an `IterationTypes`
+         * record is returned. Otherwise, `undefined` is returned.
+         */
+        function getIterationTypesOfIterator(type: Type, resolver: IterationTypesResolver, errorNode: Node | undefined) {
+            if (isTypeAny(type)) {

```

Comment:
Shouldn't `never` likewise implicitly confirm to the iterator interface (taking the most permissive types for each position), rather than being a `noIterationTypes` result? Or would doing so create a worse set of errors when you attempt to iterate over a `never`?

Reply:
This doesn't change from our previous approach. We currently don't allow you to `for..of` or spread a `never`.